From: Hannes Reinecke <hare@suse.de>
Date: Wed, 15 Jan 2014 08:11:24 +0100
Subject: hpsa: marking older boards as unsupported
Patch-Mainline: never
References: FATE#316683

While the hpsa driver works with older boards, such a configuration
is unsupported. So issue a warning in these cases.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 8e1f323..6a92881 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -4431,6 +4431,10 @@ static int hpsa_lookup_board_id(struct pci_dev *pdev, u32 *board_id)
 			"0x%08x, ignoring.\n", *board_id);
 			return -ENODEV;
 	}
+#ifdef CONFIG_SUSE_KERNEL_SUPPORTED
+	add_taint(TAINT_NO_SUPPORT, LOCKDEP_STILL_OK);
+#endif
+	dev_warn(&pdev->dev, "unsupported board ID: 0x%08x\n", *board_id);
 	return ARRAY_SIZE(products) - 1; /* generic unknown smart array */
 }
 
